File: src/main/java/org/cobbzilla/util/io/FileSuffixFilter.java
package org.cobbzilla.util.io;

import java.io.File;
import java.io.FileFilter;

public class FileSuffixFilter implements FileFilter {

    private String suffix;

    @java.beans.ConstructorProperties({"suffix"})
    public FileSuffixFilter(String suffix) {
        this.suffix = suffix;
    }

    @Override public boolean accept(File pathname) { return pathname.getName().endsWith(suffix); }

    public String getSuffix() {
        return this.suffix;
    }

    public void setSuffix(String suffix) {
        this.suffix = suffix;
    }
}

File: src/main/java/org/cobbzilla/util/io/FileSuffixFilter.kt
package org.cobbzilla.util.io

import java.io.File
import java.io.FileFilter

class FileSuffixFilter @java.beans.ConstructorProperties("suffix")
constructor(var suffix: String?) : FileFilter {

    override fun accept(pathname: File): Boolean {
        return pathname.name.endsWith(suffix!!)
    }
}
